In [1]:
import numpy as np
import os
import sys
import pandas as pd
import seaborn as sns
from pathlib import Path
import re
from datetime import datetime
from scipy.stats import mode

In [34]:
df = pd.read_csv('interim_data_process_2.csv' , dtype= {'msisdn': 'str'})

In [35]:
df.columns

Index(['msisdn', 'nbre_call_entrant_inter', 'nbre_call_sortant_inter',
       'duree_entrant_inter', 'duree_sortant_inter', 'duree_total_inter',
       'duree_moy_sortant_inter', 'duree_moy_inter', 'nbr_cor_dist',
       'nbr_appel_abs_entr_moy', 'nbr_appel_abs_entr_med',
       'nbr_appel_abs_entr_tot', 'nbr_entrant_distinct_absence_tot',
       'nbr_entrant_distinct_absence_med', 'nbr_appel_abs_sort_moy',
       'nbr_appel_abs_sort_med', 'nbr_appel_abs_sort_tot', 'maxtcpsession',
       'nbyte', 'nbytetcp', 'nbyteudp', 'byteweb', 'bytep2p', 'bytedownload',
       'bytenews', 'bytemail', 'byteothers', 'bytecontrol', 'bytegames',
       'bytestreaming', 'bytechat', 'bytevoip', 'packvpn', 'bytevvm',
       'bytestreamavsp', 'nbre_transaction_tot', 'nbre_transaction_med',
       'nbre_distinct_msisdn_tot', 'nbre_distinct_msisdn_med', 'duree_tot',
       'duree_med', 'nbre_distinct_imei', 'nombre_transaction_erecharge',
       'montant_erecharge', 'nbre_imsi', 'montant', 'montant_moy',
  

In [36]:
df.head()

msisdn  nbre_call_entrant_inter  nbre_call_sortant_inter  \
0  67969926                      5.0                      0.0   
1  88387117                      NaN                      NaN   
2  79023078                      NaN                      NaN   
3  08784993                      2.0                      0.0   
4  78703184                      NaN                      NaN   

   duree_entrant_inter  duree_sortant_inter  duree_total_inter  \
0                145.0                  0.0              145.0   
1                  NaN                  NaN                NaN   
2                  NaN                  NaN                NaN   
3                 79.0                  0.0               79.0   
4                  NaN                  NaN                NaN   

   duree_moy_sortant_inter  duree_moy_inter  nbr_cor_dist  \
0                      0.0             93.0           1.0   
1                      NaN              NaN           NaN   
2                      NaN              NaN           NaN   
3                      0.0             46.0           1.0   
4                      NaN              NaN           NaN   

   nbr_appel_abs_entr_moy     ...      duree_moy_call_complice  \
0                     1.0     ...                          NaN   
1                     NaN     ...                        100.0   
2                     NaN     ...                          NaN   
3                     1.0     ...                        169.0   
4                     NaN     ...                          NaN   

   volume_data_total  volume_data_moy  volume_data_med  smsisdn  \
0                NaN              NaN              NaN      NaN   
1       8.493466e+09     3.860666e+08      411784192.0      NaN   
2                NaN              NaN              NaN      NaN   
3       1.576696e+08     1.970871e+07       11199892.5      NaN   
4                NaN              NaN              NaN      NaN   

   nbre_corr_sortant  nbre_corr_sortant_distinc  nbre_corr_entrant  \
0                NaN                        NaN                NaN   
1                NaN                        NaN                NaN   
2                NaN                        NaN                NaN   
3                NaN                        NaN                NaN   
4                NaN                        NaN                NaN   

   nbre_corr_entrant_distinct  taux_inter  
0                         NaN         NaN  
1                         NaN         NaN  
2                         NaN         NaN  
3                         NaN         NaN  
4                         NaN         NaN  

[5 rows x 96 columns]

In [37]:
del df['smsisdn']

In [38]:
df.head()

msisdn  nbre_call_entrant_inter  nbre_call_sortant_inter  \
0  67969926                      5.0                      0.0   
1  88387117                      NaN                      NaN   
2  79023078                      NaN                      NaN   
3  08784993                      2.0                      0.0   
4  78703184                      NaN                      NaN   

   duree_entrant_inter  duree_sortant_inter  duree_total_inter  \
0                145.0                  0.0              145.0   
1                  NaN                  NaN                NaN   
2                  NaN                  NaN                NaN   
3                 79.0                  0.0               79.0   
4                  NaN                  NaN                NaN   

   duree_moy_sortant_inter  duree_moy_inter  nbr_cor_dist  \
0                      0.0             93.0           1.0   
1                      NaN              NaN           NaN   
2                      NaN              NaN           NaN   
3                      0.0             46.0           1.0   
4                      NaN              NaN           NaN   

   nbr_appel_abs_entr_moy     ...      duree  duree_moy_call_complice  \
0                     1.0     ...        NaN                      NaN   
1                     NaN     ...      200.0                    100.0   
2                     NaN     ...        NaN                      NaN   
3                     1.0     ...      676.0                    169.0   
4                     NaN     ...        NaN                      NaN   

   volume_data_total  volume_data_moy  volume_data_med  nbre_corr_sortant  \
0                NaN              NaN              NaN                NaN   
1       8.493466e+09     3.860666e+08      411784192.0                NaN   
2                NaN              NaN              NaN                NaN   
3       1.576696e+08     1.970871e+07       11199892.5                NaN   
4                NaN              NaN              NaN                NaN   

   nbre_corr_sortant_distinc  nbre_corr_entrant  nbre_corr_entrant_distinct  \
0                        NaN                NaN                         NaN   
1                        NaN                NaN                         NaN   
2                        NaN                NaN                         NaN   
3                        NaN                NaN                         NaN   
4                        NaN                NaN                         NaN   

   taux_inter  
0         NaN  
1         NaN  
2         NaN  
3         NaN  
4         NaN  

[5 rows x 95 columns]

In [39]:
df.columns

Index(['msisdn', 'nbre_call_entrant_inter', 'nbre_call_sortant_inter',
       'duree_entrant_inter', 'duree_sortant_inter', 'duree_total_inter',
       'duree_moy_sortant_inter', 'duree_moy_inter', 'nbr_cor_dist',
       'nbr_appel_abs_entr_moy', 'nbr_appel_abs_entr_med',
       'nbr_appel_abs_entr_tot', 'nbr_entrant_distinct_absence_tot',
       'nbr_entrant_distinct_absence_med', 'nbr_appel_abs_sort_moy',
       'nbr_appel_abs_sort_med', 'nbr_appel_abs_sort_tot', 'maxtcpsession',
       'nbyte', 'nbytetcp', 'nbyteudp', 'byteweb', 'bytep2p', 'bytedownload',
       'bytenews', 'bytemail', 'byteothers', 'bytecontrol', 'bytegames',
       'bytestreaming', 'bytechat', 'bytevoip', 'packvpn', 'bytevvm',
       'bytestreamavsp', 'nbre_transaction_tot', 'nbre_transaction_med',
       'nbre_distinct_msisdn_tot', 'nbre_distinct_msisdn_med', 'duree_tot',
       'duree_med', 'nbre_distinct_imei', 'nombre_transaction_erecharge',
       'montant_erecharge', 'nbre_imsi', 'montant', 'montant_moy',
  

In [40]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):  # more options can be specified also
    print(df.isnull().sum())

msisdn                                    1
nbre_call_entrant_inter              968316
nbre_call_sortant_inter              968316
duree_entrant_inter                  968316
duree_sortant_inter                  968316
duree_total_inter                    968316
duree_moy_sortant_inter              968316
duree_moy_inter                      968316
nbr_cor_dist                         968316
nbr_appel_abs_entr_moy              1385484
nbr_appel_abs_entr_med              1385484
nbr_appel_abs_entr_tot              1385484
nbr_entrant_distinct_absence_tot    1385484
nbr_entrant_distinct_absence_med    1385484
nbr_appel_abs_sort_moy              1305822
nbr_appel_abs_sort_med              1305822
nbr_appel_abs_sort_tot              1305822
maxtcpsession                       1168498
nbyte                               1168498
nbytetcp                            1168498
nbyteudp                            1168498
byteweb                             1168498
bytep2p                         

In [41]:
df.shape

(1765409, 95)

In [42]:
df_NA = df.isnull().sum().to_frame()

In [43]:
df_NA.columns = ['total']

In [44]:
df_NA = df_NA.reset_index()

In [45]:
df_NA.columns = ['var', 'total']

In [46]:
df_NA['taux_valeur_manquante'] = df_NA['total']/df.shape[0]

In [47]:
df_NA['taux_valeur_manquante'] = df_NA['taux_valeur_manquante'] * 100

In [92]:
df_NA.head()

var   total  taux_valeur_manquante
0                   msisdn       1               0.000057
1  nbre_call_entrant_inter  968316              54.849386
2  nbre_call_sortant_inter  968316              54.849386
3      duree_entrant_inter  968316              54.849386
4      duree_sortant_inter  968316              54.849386

In [73]:
df_NA_  = df_NA[df_NA['taux_valeur_manquante'] > 80]

In [74]:
df_NA_

var    total  taux_valeur_manquante
35        nbre_transaction_tot  1670174              94.605499
36        nbre_transaction_med  1670174              94.605499
37    nbre_distinct_msisdn_tot  1670174              94.605499
38    nbre_distinct_msisdn_med  1670174              94.605499
39                   duree_tot  1670174              94.605499
40                   duree_med  1670174              94.605499
41          nbre_distinct_imei  1644350              93.142722
45                     montant  1639553              92.871000
46                 montant_moy  1639553              92.871000
48      volumepartage_data_med  1760728              99.734849
49            nbr_partage_data  1760728              99.734849
50          total_partage_data  1760728              99.734849
81      volume_family_data_med  1763787              99.908123
82             nbr_family_data  1763787              99.908123
83             family_data_tot  1763787              99.908123
84          nbre_call_complice  1426973              80.829598
85                       duree  1426973              80.829598
86     duree_moy_call_complice  1426973              80.829598
90           nbre_corr_sortant  1739150              98.512583
91   nbre_corr_sortant_distinc  1739150              98.512583
92           nbre_corr_entrant  1764881              99.970092
93  nbre_corr_entrant_distinct  1764881              99.970092
94                  taux_inter  1764881              99.970092

In [66]:
df_NA_.shape

(20, 3)

In [54]:
df_ = df[ [x for x in df.columns if x not in set(df_NA_['var'])]]

In [84]:
df_.head(10)

msisdn  nbre_call_entrant_inter  nbre_call_sortant_inter  \
0  67969926                      5.0                      0.0   
1  88387117                      0.0                      0.0   
2  79023078                      0.0                      0.0   
3  08784993                      2.0                      0.0   
4  78703184                      0.0                      0.0   
5  58568917                      0.0                      0.0   
6  07370812                      6.0                      0.0   
7  57510157                      7.0                      0.0   
8  09884891                      1.0                      0.0   
9  47893429                      4.0                      0.0   

   duree_entrant_inter  duree_sortant_inter  duree_total_inter  \
0                145.0                  0.0              145.0   
1                  0.0                  0.0                0.0   
2                  0.0                  0.0                0.0   
3                 79.0                  0.0               79.0   
4                  0.0                  0.0                0.0   
5                  0.0                  0.0                0.0   
6                239.0                  0.0              239.0   
7                101.0                  0.0              101.0   
8                109.0                  0.0              109.0   
9                400.0                  0.0              400.0   

   duree_moy_sortant_inter  duree_moy_inter  nbr_cor_dist  \
0                      0.0             93.0           1.0   
1                      0.0              0.0           0.0   
2                      0.0              0.0           0.0   
3                      0.0             46.0           1.0   
4                      0.0              0.0           0.0   
5                      0.0              0.0           0.0   
6                      0.0            104.0           1.0   
7                      0.0             20.2           1.0   
8                      0.0            109.0           1.0   
9                      0.0            400.0           1.0   

   nbr_appel_abs_entr_moy       ...         nbr_corr_dist_sortant_sms_med  \
0                1.000000       ...                                   1.0   
1                0.000000       ...                                   2.0   
2                0.000000       ...                                   2.0   
3                1.000000       ...                                   2.0   
4                0.000000       ...                                   2.0   
5                0.000000       ...                                   2.0   
6                0.000000       ...                                   1.0   
7                0.000000       ...                                   5.0   
8                2.000000       ...                                  34.0   
9                1.166667       ...                                 302.0   

   nbr_duree_data_tot  nbr_duree_data_med  nbr_duree_data_moy  \
0                 0.0                 0.0            0.000000   
1           1707645.0              2263.0         5143.509036   
2                 0.0                 0.0            0.000000   
3             78189.0              1455.0         2522.225806   
4                 0.0                 0.0            0.000000   
5                 0.0                 0.0            0.000000   
6            327524.0               327.0          519.055468   
7                 0.0                 0.0            0.000000   
8           1511825.0              1559.0        45812.878788   
9           3783936.0              2690.0         8279.947484   

   nbre_call_complice   duree  duree_moy_call_complice  volume_data_total  \
0                 0.0     0.0                     0.00       0.000000e+00   
1                 3.0   200.0                   100.00       8.493466e+09   
2                 0.0     0.0                     0.00       0.000000e+00   
3                1

In [59]:
df_ = df_.drop(index = 1765408)

In [83]:
df_.head(10)

msisdn  nbre_call_entrant_inter  nbre_call_sortant_inter  \
0  67969926                      5.0                      0.0   
1  88387117                      0.0                      0.0   
2  79023078                      0.0                      0.0   
3  08784993                      2.0                      0.0   
4  78703184                      0.0                      0.0   
5  58568917                      0.0                      0.0   
6  07370812                      6.0                      0.0   
7  57510157                      7.0                      0.0   
8  09884891                      1.0                      0.0   
9  47893429                      4.0                      0.0   

   duree_entrant_inter  duree_sortant_inter  duree_total_inter  \
0                145.0                  0.0              145.0   
1                  0.0                  0.0                0.0   
2                  0.0                  0.0                0.0   
3                 79.0                  0.0               79.0   
4                  0.0                  0.0                0.0   
5                  0.0                  0.0                0.0   
6                239.0                  0.0              239.0   
7                101.0                  0.0              101.0   
8                109.0                  0.0              109.0   
9                400.0                  0.0              400.0   

   duree_moy_sortant_inter  duree_moy_inter  nbr_cor_dist  \
0                      0.0             93.0           1.0   
1                      0.0              0.0           0.0   
2                      0.0              0.0           0.0   
3                      0.0             46.0           1.0   
4                      0.0              0.0           0.0   
5                      0.0              0.0           0.0   
6                      0.0            104.0           1.0   
7                      0.0             20.2           1.0   
8                      0.0            109.0           1.0   
9                      0.0            400.0           1.0   

   nbr_appel_abs_entr_moy       ...         nbr_corr_dist_sortant_sms_med  \
0                1.000000       ...                                   1.0   
1                0.000000       ...                                   2.0   
2                0.000000       ...                                   2.0   
3                1.000000       ...                                   2.0   
4                0.000000       ...                                   2.0   
5                0.000000       ...                                   2.0   
6                0.000000       ...                                   1.0   
7                0.000000       ...                                   5.0   
8                2.000000       ...                                  34.0   
9                1.166667       ...                                 302.0   

   nbr_duree_data_tot  nbr_duree_data_med  nbr_duree_data_moy  \
0                 0.0                 0.0            0.000000   
1           1707645.0              2263.0         5143.509036   
2                 0.0                 0.0            0.000000   
3             78189.0              1455.0         2522.225806   
4                 0.0                 0.0            0.000000   
5                 0.0                 0.0            0.000000   
6            327524.0               327.0          519.055468   
7                 0.0                 0.0            0.000000   
8           1511825.0              1559.0        45812.878788   
9           3783936.0              2690.0         8279.947484   

   nbre_call_complice   duree  duree_moy_call_complice  volume_data_total  \
0                 0.0     0.0                     0.00       0.000000e+00   
1                 3.0   200.0                   100.00       8.493466e+09   
2                 0.0     0.0                     0.00       0.000000e+00   
3                1

In [61]:
df_ = df_.fillna(0)

In [70]:
df_.to_csv("data_process.csv", index = False)

In [64]:
def forte_corr(seuil = 0.5 , df = df_ , method = 'pearson') :
    #list_plot = df.columns[2:]
    list_plot = df.columns
    L_index = []
    List_var = []
    List_corr = []
    
    cor = df[list_plot].corr(method = method)
    cor = cor[cor > seuil]
    index = cor.index
    for col in cor.columns : 
        for indice , row in enumerate(cor[col]):
            if (not np.isnan(row)) and (row != 1 ) :
                L_index.append(index[indice])
                List_var.append(col)
                List_corr.append(row)
    #[(x , y , z) for x , y , z in zip (L_index , List_var ,List_corr )]                                                    
    df = pd.DataFrame({'Var1':L_index , 'Var2':List_var , 'Correlation':List_corr}).sort_values('Correlation').reset_index()
    df = df.drop('index' , 1)
    
    return df[[True if x%2 == 0 else False for x in range(df.shape[0])]]

In [67]:
df_corr = forte_corr()

In [82]:
df_corr

Var1                              Var2  \
0       nbr_appel_total_sortant_onnet            nbr_appel_abs_sort_med   
2                  nbr_duree_data_tot     nbr_corr_dist_entrant_sms_med   
4                  nbr_duree_data_tot     nbr_corr_dist_entrant_sms_tot   
6                       bytestreaming                   volume_data_moy   
8       nbr_corr_dist_entrant_sms_med                      nbr_cor_dist   
10            duree_sortant_moy_ofnet         duree_total_sortant_ofnet   
12        nbr_appel_sortant_moy_ofnet         duree_total_sortant_ofnet   
14      nbr_corr_dist_entrant_sms_tot                      nbr_cor_dist   
16      nbr_corr_dist_entrant_sms_med            nbr_appel_abs_sort_med   
18             nbr_appel_abs_sort_med     nbr_corr_dist_entrant_sms_tot   
20      nbr_corr_dist_entrant_sms_med       nbr_sms_total_sortant_onnet   
22                    volume_data_moy                          bytechat   
24      nbr_corr_dist_entrant_sms_tot       nbr_sms_total_sortant_onnet   
26                 nbr_duree_data_tot                   volume_data_moy   
28                            byteweb                      bytedownload   
30        nbr_appel_sortant_med_onnet                      nbr_cor_dist   
32                       nbr_cor_dist           duree_sortant_med_onnet   
34             nbr_appel_abs_sort_med           duree_sortant_moy_ofnet   
36        nbr_appel_sortant_moy_ofnet            nbr_appel_abs_sort_med   
38             nbr_appel_abs_sort_moy           duree_sortant_moy_ofnet   
40            duree_sortant_moy_ofnet            nbr_appel_abs_sort_moy   
42             nbr_appel_abs_sort_moy     nbr_corr_dist_entrant_sms_med   
44      nbr_appel_total_sortant_onnet            nbr_appel_abs_sort_moy   
46      nbr_appel_total_sortant_onnet           duree_sortant_moy_onnet   
48      nbr_appel_total_sortant_onnet       nbr_appel_sortant_moy_onnet   
50             nbr_appel_abs_sort_moy     nbr_corr_dist_entrant_sms_tot   
52            duree_sortant_moy_ofnet     nbr_appel_total_sortant_ofnet   
54        nbr_appel_sortant_moy_ofnet     nbr_appel_total_sortant_ofnet   
56             nbr_appel_abs_sort_tot     nbr_appel_total_sortant_onnet   
58      nbr_appel_total_sortant_ofnet     nbr_corr_dist_entrant_sms_med   
..                                ...                               ...   
256                             nbyte                     bytestreaming   
258                             duree         duree_total_sortant_onnet   
260         duree_total_sortant_ofnet     nbr_appel_total_sortant_ofnet   
262                   volume_data_moy                 volume_data_total   
264                           byteweb                             nbyte   
266               duree_sortant_inter           nbre_call_sortant_inter   
268                             duree           duree_moy_call_complice   
270                          nbyteudp                             nbyte   
272                nbr_duree_data_med                nbr_duree_data_moy   
274                          nbyteudp                           byteweb   
276           duree_sortant_moy_onnet           duree_sortant_med_onnet   
278       nbr_appel_sortant_moy_onnet       nbr_appel_sortant_med_onnet   
280           duree_sortant_med_onnet       nbr_appel_sortant_moy_onnet   
282           duree_sortant_moy_onnet       nbr_appel_sortant_med_onnet   
284       nbr_appel_sortant_med_ofnet       nbr_appel_sortant_moy_ofnet   
286           duree_sortant_med_ofnet       nbr_appel_sortant_moy_ofnet   
288           duree_sortant_moy_ofnet           duree_sortant_med_ofnet   
290       nbr_appel_sortant_med_ofnet           duree_sortant_moy_ofnet   
292         nbr_sms_sortant_moy_onnet         nbr_sms_sortant_med_onnet   
294                   volume_data_med                   volume_data_moy   
296         nbr_sms_sortant_moy_ofnet         nbr_sms_sortant_med_ofnet   
298                          nbytetcp                     

In [75]:
df_corr_test_f = df_corr[df_corr['Correlation']  > 0.8 ]

In [86]:
df_corr_test_f

Var1                              Var2  \
250                nbre_call_complice                             duree   
252                          nbytetcp                     bytestreaming   
254                   volume_data_med                 volume_data_total   
256                             nbyte                     bytestreaming   
258                             duree         duree_total_sortant_onnet   
260         duree_total_sortant_ofnet     nbr_appel_total_sortant_ofnet   
262                   volume_data_moy                 volume_data_total   
264                           byteweb                             nbyte   
266               duree_sortant_inter           nbre_call_sortant_inter   
268                             duree           duree_moy_call_complice   
270                          nbyteudp                             nbyte   
272                nbr_duree_data_med                nbr_duree_data_moy   
274                          nbyteudp                           byteweb   
276           duree_sortant_moy_onnet           duree_sortant_med_onnet   
278       nbr_appel_sortant_moy_onnet       nbr_appel_sortant_med_onnet   
280           duree_sortant_med_onnet       nbr_appel_sortant_moy_onnet   
282           duree_sortant_moy_onnet       nbr_appel_sortant_med_onnet   
284       nbr_appel_sortant_med_ofnet       nbr_appel_sortant_moy_ofnet   
286           duree_sortant_med_ofnet       nbr_appel_sortant_moy_ofnet   
288           duree_sortant_moy_ofnet           duree_sortant_med_ofnet   
290       nbr_appel_sortant_med_ofnet           duree_sortant_moy_ofnet   
292         nbr_sms_sortant_moy_onnet         nbr_sms_sortant_med_onnet   
294                   volume_data_med                   volume_data_moy   
296         nbr_sms_sortant_moy_ofnet         nbr_sms_sortant_med_ofnet   
298                          nbytetcp                             nbyte   
300            nbr_appel_abs_sort_moy            nbr_appel_abs_sort_med   
302  nbr_entrant_distinct_absence_med            nbr_appel_abs_entr_moy   
304            nbr_appel_abs_entr_med            nbr_appel_abs_entr_moy   
306  nbr_entrant_distinct_absence_tot            nbr_appel_abs_entr_tot   
308     nbr_corr_dist_entrant_sms_tot     nbr_corr_dist_entrant_sms_med   
310               duree_entrant_inter                 duree_total_inter   
312            nbr_appel_abs_entr_med  nbr_entrant_distinct_absence_med   
314     nbr_corr_dist_sortant_sms_med     nbr_corr_dist_sortant_sms_tot   

     Correlation  
250     0.801996  
252     0.822171  
254     0.828737  
256     0.831476  
258     0.840948  
260     0.851725  
262     0.860758  
264     0.863344  
266     0.880310  
268     0.887582  
270     0.891112  
272     0.892106  
274     0.896015  
276     0.914967  
278     0.914967  
280     0.914967  
282     0.914967  
284     0.935492  
286     0.935492  
288     0.935492  
290     0.935492  
292     0.937443  
294     0.961697  
296     0.965334  
298     0.966401  
300     0.981017  
302     0.992350  
304     0.993143  
306     0.997961  
308     0.998046  
310     0.998277  
312     0.999121  
314     0.999977

In [77]:
df_corr_test_f.columns

Index(['Var1', 'Var2', 'Correlation'], dtype='object')

In [78]:
df_corr_test_f['Var1'].nunique()

26

In [79]:
df_corr_test_f['Var2'].nunique()

26

In [87]:
print(set(df_corr_test_f['Var1'].unique()) | set(df_corr_test_f['Var2'].unique()))

{'duree_sortant_moy_ofnet', 'duree_sortant_med_onnet', 'nbr_corr_dist_sortant_sms_tot', 'volume_data_moy', 'nbr_appel_sortant_moy_ofnet', 'volume_data_total', 'duree', 'nbr_sms_sortant_moy_ofnet', 'nbr_appel_sortant_med_onnet', 'nbr_appel_sortant_moy_onnet', 'duree_entrant_inter', 'nbr_corr_dist_entrant_sms_med', 'nbr_duree_data_med', 'nbre_call_sortant_inter', 'nbr_sms_sortant_med_onnet', 'nbyte', 'nbr_appel_abs_entr_tot', 'nbr_appel_sortant_med_ofnet', 'nbytetcp', 'duree_sortant_inter', 'nbr_entrant_distinct_absence_med', 'nbr_duree_data_moy', 'byteweb', 'nbr_sms_sortant_med_ofnet', 'nbr_sms_sortant_moy_onnet', 'nbr_appel_abs_entr_med', 'duree_moy_call_complice', 'duree_total_inter', 'volume_data_med', 'duree_total_sortant_onnet', 'duree_sortant_moy_onnet', 'bytestreaming', 'duree_sortant_med_ofnet', 'duree_total_sortant_ofnet', 'nbyteudp', 'nbr_appel_total_sortant_ofnet', 'nbr_appel_abs_sort_med', 'nbr_appel_abs_entr_moy', 'nbre_call_complice', 'nbr_corr_dist_sortant_sms_med', 'nbr_

In [88]:
len(set(df_corr_test_f['Var1'].unique()) | set(df_corr_test_f['Var2'].unique()))

43

__________

# NA management

In [93]:
with pd.option_context('display.max_rows', None, 'display.max_columns', None):  # more options can be specified also
    print(df_.isnull().sum())

msisdn                              0
nbre_call_entrant_inter             0
nbre_call_sortant_inter             0
duree_entrant_inter                 0
duree_sortant_inter                 0
duree_total_inter                   0
duree_moy_sortant_inter             0
duree_moy_inter                     0
nbr_cor_dist                        0
nbr_appel_abs_entr_moy              0
nbr_appel_abs_entr_med              0
nbr_appel_abs_entr_tot              0
nbr_entrant_distinct_absence_tot    0
nbr_entrant_distinct_absence_med    0
nbr_appel_abs_sort_moy              0
nbr_appel_abs_sort_med              0
nbr_appel_abs_sort_tot              0
maxtcpsession                       0
nbyte                               0
nbytetcp                            0
nbyteudp                            0
byteweb                             0
bytep2p                             0
bytedownload                        0
bytenews                            0
bytemail                            0
byteothers  

______________________

# PCA

In [95]:
from sklearn.preprocessing import StandardScaler
features = df_.columns
x = df_.loc[:, features[1:]].values# Separating out the target
x = StandardScaler().fit_transform(x)

In [96]:
from sklearn.decomposition import PCA

In [98]:
pca = PCA()

principalComponents = pca.fit_transform(x)

In [101]:
# Nombre de composantes
print(pca.n_components_)

74


# Conlusion 

- Pour notre étude nous avons *1765409 individus et 95 variables*
- 20 variables ayant des valeurs manquantes à plus de **94%**, elles ont fait objets de suppression
- Les variables étant toutes quantitatives et NA représentant l'inactivité de l'individu par rapport la variable concerner, les valeurs manquantes ont toutes été remplacé par **0**.
- 43 variables correleés à plus de **80%**
- D'après l'analyse en composante principale les dim1 et 2 représente **32%** de l'initie total ce qui est quand meme important vu le nombre de variables soit 75.
- Nous avons pu dégager 2 groupes de variables :

  - premier groupe de variables:
  
    Ce groupe est lié au trafic appels et sms

  `nombre_distinct_profil`          
  `nbr_sms_sortant_med_ofnet`        
  `nbr_sms_sortant_med_onnet`
  `nbre_imsi`        
 `nbre_call_entrant_inter`           
 `nbre_call_sortant_inter`           
 `duree_entrant_inter`               
 `duree_sortant_inter`               
 `duree_total_inter`                 
 `duree_moy_sortant_inter`           
 `duree_moy_inter`                   
 `nbr_cor_dist`                      
 `nbr_appel_abs_entr_moy`            
 `nbr_appel_abs_entr_med`            
 `nbr_appel_abs_entr_tot`            
 `nbr_entrant_distinct_absence_tot`  
 `nbr_entrant_distinct_absence_med`  
 `nbr_appel_abs_sort_moy`            
 `nbr_appel_abs_sort_med `           
 `nbr_appel_abs_sort_tot`            
 `nbre_call_complice `               
 `duree `                            
 `duree_moy_call_complice `          
 `nombre_transaction_erecharge `     
 `nbr_appel_sortant_moy_ofnet`      
 `nbr_appel_total_sortant_ofnet`     
 `nbr_appel_sortant_med_ofnet `      
 `duree_total_sortant_ofnet `        
 `duree_sortant_med_ofnet  `         
 `duree_sortant_moy_ofnet  `        
 `nbr_appel_sortant_moy_onnet `      
 `nbr_appel_total_sortant_onnet`     
 `nbr_appel_sortant_med_onnet`      
 `duree_total_sortant_onnet`         
 `duree_sortant_med_onnet`         
 `duree_sortant_moy_onnet`        
 `nbr_rech_sos`                    
 `nbr_moy_sos`                       
 `montant_moy_sos`                   
 `nbr_corr_dist_entrant_sms_tot`     
 `nbr_corr_dist_entrant_sms_med` 
 
  - Deuxième groupe de variables :
  
    Ce ensemble de variable est lié à la data 

 `maxtcpsession`                     
 `nbyte    `                        
` nbytetcp `                         
` nbyteudp `                         
` byteweb `                        
 `bytep2p `                        
` bytedownload `                    
` bytenews `                      
` bytemail `                         
` byteothers `                       
 `bytecontrol `                      
` bytegames `                        
` bytestreaming `                    
` bytechat`                         
 `bytevoip `                        
 `packvpn `                        
 `bytevvm `                       
 `bytestreamavsp   `                 
 `montant_erecharge `                
 `nbr_om_transact `                 
 `montant_tot_sos`                   
`nbr_sms_sortant_moy_ofnet`        
` nbr_sms_total_sortant_ofnet `      
 `nbr_sms_sortant_moy_onnet `        
 `nbr_sms_total_sortant_onnet  `    
 `nbr_corr_dist_sortant_sms_tot`     
 `nbr_corr_dist_sortant_sms_med `    
 `nbr_duree_data_tot `             
 `nbr_duree_data_med`                
` nbr_duree_data_moy `             
 `volume_data_total`                 
 `volume_data_moy`                 
 `volume_data_med`  
 


# Conlusion SMILE 

- Pour notre étude nous avons *1761756 individus et 74 variables*
- 72 variables quantitatives et 1 variable qualitatite, la variable restante est le $msisdn$ .
- Nous avons orienté notre analyse vers certaines variables.
   - nombre_appels_nuit
   - numb_call_inter
   - nombre_appels

   
   
  - **nombre_appels_nuit**
  
<dd>NOMBRE DE LIGNES NON-NULLES:94,785<dd>
<dd>25%:1<dd>
<dd>50%:1<dd>
<dd>75%:3<dd>
<dd>MAX:144<dd>
<dd>MOYENNE:2.2831<dd>
<dd>MEDIAN:1<dd>
<dd>MIN:1<dd>
<dd>VAR:6.6904<dd>
<dd>CORRELATION:  correlée avec aucune variable, a part nombre_appels_nuit_distinct  <dd>

   - **numb_call_inter**
  
<dd>NOMBRE DE LIGNES NON-NULLES:1,583<dd>
<dd>25%:4<dd>
<dd>50%:9<dd>
<dd>75%:18<dd>
<dd>MAX:350<dd>
<dd>MOYENNE:14.8484<dd>
<dd>MEDIAN:9<dd>
<dd>MIN:2<dd>
<dd>VAR:420.6546<dd>
<dd>CORRELATION:  Tres corrélée avec toutes les autres variables internationales  <dd>

   - **nombre_appels** 

<dd>NOMBRE DE LIGNES NON-NULLES:815,470<dd>
<dd>25%:19<dd>
<dd>50%:70<dd>
<dd>75%:195<dd>
<dd>MAX:3,814<dd>
<dd>MOYENNE:147.8024<dd>
<dd>MEDIAN:70<dd>
<dd>MIN:1<dd>
<dd>VAR:41,330.4692<d&d>
<dd>CORRELATION:  Très correlée avec toutes les autres variables : nombre d'appels sortants, entrents, etc…  <dd>